src/app/pages/BlocksPage/index.tsx

```diff
@@ -1,19 +1,43 @@
-import { Link } from 'react-router-dom'
 import { FC } from 'react'
+import Card from '@mui/material/Card'
+import CardHeader from '@mui/material/CardHeader'
 import { PageLayout } from '../../components/PageLayout'
 import { useGetEmeraldBlocks } from '../../../oasis-indexer/api'
-import { emeraldRoute } from '../../../routes'
+import { useTranslation } from 'react-i18next'
+import CardContent from '@mui/material/CardContent'
+import { useParams } from 'react-router-dom'
+
+import { Blocks } from '../../components/Blocks'
+import { NUMBER_OF_ITEMS_ON_SEPARATE_PAGE } from '../../config'
+
+const PAGE_SIZE = NUMBER_OF_ITEMS_ON_SEPARATE_PAGE
 
 export const BlocksPage: FC = () => {
-  const { data } = useGetEmeraldBlocks({}, { query: { refetchInterval: 5000 } })
+  const { t } = useTranslation()
+  const { offsetString = '0' } = useParams()
```

Comment:
Can we use `useSearchParams` for getting the offset?

Reply:
I have now implemented that  in the API layer, adding a wrapper around the generated API, just like we have it for transactions.